Put all database connection strings in keyvault

eerhardt opened the pull request:
## Description

When using password authentication and child databases in Azure, don't create a connection string based on the parent's connection string (which is a secretOutputs). The reason being that the deployment tool needs to grab the parent server's connection string from the keyvault and add to it.

Instead, put the database connection strings into the keyvault as well. This way all the secrets are contained in the keyvault and they can be referenced the same way as the parent - via keyvault.

Fix #3299

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [x] Yes
- Did you add public API?
  - [x] No
- Does the change make any security assumptions or guarantees?
  - [x] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [x] No
- Does the change require an update in our Aspire docs?
  - [ ] Yes
    - Link to aspire-docs issue: 
  - [x] No

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/6059)

eerhardt committed:
Put all database connection strings in keyvault

When using password authentication and child databases in Azure, don't create a connection string based on the parent's connection string (which is a secretOutputs). The reason being that the deployment tool needs to grab the parent server's connection string from the keyvault and add to it.

Instead, put the database connection strings into the keyvault as well. This way all the secrets are contained in the keyvault and they can be referenced the same way as the parent - via keyvault.

Fix #3299

eerhardt committed:
Clean up code structure

davidfowl commented:
Can you change this test https://github.com/dotnet/aspire/blob/d13467e2c020155aba2b6e29add2f0bb7228129d/tests/Aspire.Hosting.Azure.Tests/AzureContainerAppsTests.cs#L249

eerhardt committed:
Update secret name to be a valid KeyVault secret name.

Since KV secret names can't have underscores, but can have hyphens, we use a hyphen. But this means we need to sanitize the KV secret names when using them in a bicep identifier.

eerhardt committed:
Finish up

eerhardt commented on src/Aspire.Hosting.Azure/Utils/BicepIdentifierHelpers.cs:
```diff
@@ -0,0 +1,61 @@
+{
+    // See rules from Bicep's highlightjs implementation:
+    // https://github.com/Azure/bicep/blob/a992bdf2d4d7c5c7dec684b7d0de4db9cb260f8a/src/highlightjs/src/bicep.ts#L12
+    [GeneratedRegex("^[A-Za-z_][A-Za-z0-9_]*$")]
```
Note that this regex changed to no longer allow the Kelvin Sign (which a-z, ignore case allows).

davidfowl commented on src/Aspire.Hosting.Azure/Utils/BicepIdentifierHelpers.cs:
```diff
@@ -0,0 +1,61 @@
+
+namespace Aspire.Hosting.Azure.Utils;
+
+internal static partial class BicepIdentifierHelpers
```
cc @captainsafia
